torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -1432,10 +1444,61 @@ def codegen_multi_output(self, name, value):
             super().codegen_multi_output(name, value)
 
     def codegen_subgraph(self, subgraph, outer_inputs, outer_outputs):
-        raise NotImplementedError("Control flow NYI in C++ wrapper codegen.")
+        self.writeline(f"{self.comment} subgraph: {subgraph.name}")
+        for inner_input, outer_input in zip(subgraph.graph.graph_inputs, outer_inputs):
+            if config.abi_compatible:
+                self.writeline(f"AtenTensorHandle {inner_input}_handle;")
+                self.writeline(
+                    f"AOTI_TORCH_ERROR_CODE_CHECK(aoti_torch_copy_tensor({outer_input}, &{inner_input}_handle));"
+                )
+                self.writeline(
+                    f"RAIIAtenTensorHandle {inner_input}({inner_input}_handle);"
+                )
+            else:
+                self.writeline(
+                    f"{self.declare}{inner_input} = {outer_input}{self.ending}"
+                )
+        parent_graph = V.graph
+        with V.set_graph_handler(subgraph.graph):
+            subgraph.graph.codegen_subgraph(
+                parent_graph=parent_graph,
+            )
+        for inner_output, outer_output in zip(
+            subgraph.graph.graph_outputs, outer_outputs
+        ):
+            src = inner_output.codegen_reference()
+            if config.abi_compatible:
+                src = f"std::move({src})"
+            self.writeline(f"{outer_output} = {src}{self.ending}")
 
     def codegen_conditional(self, conditional):
-        raise NotImplementedError("Control flow NYI in C++ wrapper codegen.")
+        name = conditional.get_name()
```

Comment:
Do you have an example of what the output code looks like?

Reply:
Sure, for this model:

```python
        class SimpleCondModel(torch.nn.Module):
            def forward(self, p, a, b):
                def true_fn(x, y):
                    return x + y

                def false_fn(x, y):
                    return x - y

                return torch.cond(p, true_fn, false_fn, [a, b]) + a.sum()
```

Here are two versions of the output C++ codegen:

- ABI-compatible: https://gist.github.com/aakhundov/a8d14e4bc4bea7a4be388b5282afe3ea
- non-ABI-compatible: https://gist.github.com/aakhundov/88d6c7fc538bb8651e6e652ca11cb308

The full unit test is `test_aot_inductor.test_conditional_simple` in the current version of the PR.